docs/developers/parallel_transfer.md

```diff
@@ -0,0 +1,44 @@
+#
+
+## Parallel Transfer
+
+As of iRODS 4.3.4, parallel transfer via the high ports is deprecated. Future versions of iRODS will use the zone port for parallel transfer. Therefore, this page is dedicated to everything related to parallel transfer via the zone port.
+
+### Why deprecate parallel transfer over the high ports?
+
+There are a few reasons for why this functionality is deprecated:
+
+- High ports required network administrators to open a wide range of ports
+- Implementation required special code paths in the server, making it complex to maintain and reason about
+- Only provided via the Put API endpoint
+- Did not use the same implementation for secure communication
+- Added another point where client libraries could deviate from each other
+
+### What benefits are there to parallel transfer over the zone port?
+
+Given the iRODS API is POSIX-like, the iRODS Consortium wanted to investigate whether it was possible to support I/O patterns seen in POSIX environments. As it turns out, the iRODS API enabled the ability to write sparse files. This resulted in development of Logical Locking and libraries for guarding against data corruption due to uncoordinated write operations.
+
+With this functionality in place, the following benefits are provided by iRODS:
+
+- Parallel transfer no longer required special code path for secure communication
+- Removed the need for high ports
+- Removed the need for special code paths for parallel transfer
+- Reading and writing data objects is closer to what users expect from a POSIX-like system
+
+### How to perform a parallel write over the zone port
+
+To write a data object in parallel using the zone port requires some coordination between the client and the server. Parallel read operations do not require coordination.
+
+The steps are as follows:
+
+1. Open a stream to the replica of interest. We'll refer to this stream as the **primary stream**.
+1. Capture the replica access token and replica number from the **primary stream**.
+1. Open the **secondary streams**. Each stream must satisfy the following requirements:
+    - Streams must not use the same connection
+    - Streams must target the same replica
+    - Streams must use the replica access token obtained from the **primary stream**
+1. Use the streams to write to the replica.
+1. When done, close all **secondary streams** without updating the catalog.
+1. Close the **primary stream** normally.
```

Comment:
Perhaps a mention of "finalize" here would not be out of place. I know it's part of the "normal" close process, but could be good to provide that detail.

Reply:
Done.